### ISSUE TITLE
Compilers: do not print option documentation in usage messages

### ISSUE THREAD

**Issue by dbuenzli** [ocaml/ocaml#9295]
When compilers errors due to a cli error, they spit out the whole option sets with their documentation in the usage message. Nowadays this consist in 154 lines which requires to painfully scroll back to get the actual error message.

This is also not so standard. It is more customary for programs to simply output a synopsis of the cli and indicate to use `--help` for more . That is rather than: 

```
> ocamlc -bla 
ocamlc: unknown option '-bla'.
Usage: ocamlc <options> <files>
Options are:
[... 151 lines ... elided]
```
Simply output: 
```
> ocamlc -bla
ocamlc: unknown option '-bla'.
Usage: ocamlc <options> <files>
Try 'ocamlc --help' for more information.
```

A `git grep` for `"Options are:"` should guide were changes are needed.

**Comment by gasche** [comment]
Sure, why not?

**Comment by Kirti2502** [comment]
Hi, I am an outreachy applicant. Can i work on this issue?

**Comment by gasche** [comment]
Hi @Kirti2502, and thanks for being willing to contribute! This issue may not be your best choice, because Jeffa5 already proposed a pull request #9341, which is in the process of being reviewed and discussed.

**Comment by Kirti2502** [comment]
Okay, I'll take some other issue. Thanks.

**Comment by gasche** [comment]
This is still relevant and, I think, a good job for a newcomer.

**Comment by pkhry** [comment]
I added a pull request, will be happy to fix any issues with it

**Comment by gasche** [comment]
This should now be fixed now that #10289 was merged.

**Comment by dbuenzli** [comment]
Thanks @Zett98 for fixing that.
